Merge changes from Andy's branch to set up live demo

shankari opened the pull request:
This is a dup of #570. Unsure why that was ever closed.

AmplabJenkins commented:
Build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/e-mission-server-prb/1146/
Test FAILed.

shankari commented:
Documenting merge conflict resolution + additional changes for later....

Main changes were to the webapp
- removing the existing `webapp/*` and replacing with a new template based on bootstrap
  - CONFLICT: `webapp/www/js/app.js`, `webapp/www/js/controllers.js`, `webapp/www/templates/client_setup.html`: Deleted locally, modified on master
  - RESOLUTION: re-deleted
- there were also a bunch of matplotlib font cache files which clearly don't need to be rewritten (`matplotlib-ubuntu-00t8nj3u/fontList.py3k.cache`)
   - RESOLUTION: removed (`git rm -rf matplotlib-ubuntu-*`)
- bunch of .icloud files
   - RESOLUTION: removed (`find . -name \*.icloud | xargs git rm -f`)
- un-necessarily modified files
   - `Timeseries_Sample.ipynb`, `intake.stdinout`, `Classifier.ipynb`, `e-mission-py.bash`, `seed_model.json`
   - RESOLUTION: fixed them
- checked in configurations
   - RESOLUTION: removed them (`conf/log/intake.conf, ...)
- unclear errors (`e-mission-py.bash` + 2 test timelines)
   - RESOLUTION: reverted to HEAD

After all those are removed, the core changes are as below.

```
	new file:   bin/check_for_notifications.py
	new file:   bin/make_tiersys.py
	new file:   bin/update_polarbear.py
	modified:   emission/core/get_database.py
	new file:   emission/core/wrapper/happiness_change.py
	new file:   emission/core/wrapper/polarbear.py
	new file:   emission/core/wrapper/similar_trips.py
	new file:   emission/core/wrapper/suggestion_sys.py
	new file:   emission/core/wrapper/tier.py
	new file:   emission/core/wrapper/tiersys.py
	modified:   emission/core/wrapper/trip.py
	modified:   emission/core/wrapper/user.py
	modified:   emission/net/ext_service/habitica/proxy.py
	modified:   emission/pipeline/intake_stage.py
	new file:   emission/tests/TestPolarBear.py
	new file:   emission/tests/coreTests/wrapperTests/TestSuggestionSys.py
	new file:   emission/tests/coreTests/wrapperTests/TestTierSys.py
	modified:   emission/tests/coreTests/wrapperTests/TestUser.py
	new file:   emission/tests/data/real_examples/2018-01-08.2018-01-08.timeline
	new file:   responses.csv
	new file:   website_test.py
```

This is a great argument in favor of splitting out the webapp into its own repo. Many studies will want to customize it and then merging any changes gets really tricky. If the original PR (#570) had been this small, it would probably have been merged way before now 😄
